File: Programmers/기능개발.cpp
#include <string>
#include <queue>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> p, vector<int> s) {
    vector<int> answer;
    queue<int> Q;
    int tmp;

    for (int i = 0; i < p.size(); i++)
    {
        tmp = (100 - p[i]) / s[i];
        if ((100 - p[i]) % s[i] != 0)
            tmp++;
        Q.push(tmp);
    }


    while (!Q.empty())
    {
        int result = 0;
        int fr = Q.front(); Q.pop();
        result++;
        while (fr >= Q.front())
        {
            result++;
            Q.pop();
            if (Q.empty())   break;
        }
        answer.push_back(result);
    }

    return answer;
}

File: Programmers/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> Q = new LinkedList<>();
        List<Integer> pre_ans = new LinkedList<>();
     
        for(int i = 0; i < progresses.length; i++)
        {
            int tmp = (100 - progresses[i]) / speeds[i];
            if((100 - progresses[i]) % speeds[i] != 0)
                tmp++;
            Q.add(tmp);
        }
        
        while (!Q.isEmpty()) {
			int cur = Q.poll();
			int ans = 1;
            if(!Q.isEmpty())
                while (Q.peek() <= cur) {
                    Q.remove();
                    ans++;
                    if(Q.isEmpty())
                        break;
                }
			pre_ans.add(ans);

		}
        
        
        int[] answer = new int[pre_ans.size()];
        for(int i = 0; i < pre_ans.size(); i++)
            answer[i] = pre_ans.get(i);
        return answer;
    }
}